CRM/Upgrade/Incremental/php/FourSeven.php

```diff
@@ -1012,6 +1022,17 @@ public function updateKenyanProvinces(CRM_Queue_TaskContext $ctx) {
   }
 
   /**
+   * Add index to status_id field as per CRM-20313.
+   *
+   * @param \CRM_Queue_TaskContext $ctx
+   */
+  public function addIndexStatusId(CRM_Queue_TaskContext $ctx) {
+    $tables = array('civicrm_activity' => array('status_id'));
+    CRM_Core_BAO_SchemaHandler::createIndexes($tables);
+    return TRUE;
+  }
```

Comment:
@colemanw we've been recommending that function. I guess it could be moved to another class - but there are non-upgraded reasons to add & drop indexes.

My reservation about this & the 2 other open PRs for adding indexes is that on large sites adding indexes to large tables can be very slow and I feel like we should complete the work @aydun started to allow indexes to be declared & sites to reconcile them, rather than forcing them to either not upgrade to the next point version or have a significant outage.